### PR TITLE
Ecf 109977720 synchronize

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: app/scripts/app.js [hunk 1]
```diff
@@ -6,94 +15,107 @@ var lessonsRead = [];
...
-          $rootScope.$on('$routeChangeSuccess', function() {
-              if (l10n === 'es-PE') {
-                  $('body').addClass('es-PE');
-              }
-              else {
-                  $('body').removeClass('es-PE');
-              }
-          });
+        $rootScope.$on('$routeChangeSuccess', function() {
+          if (l10n === 'es-PE') {
```
Comment (michaelwehrley) [inline app/scripts/app.js]: I would make this a constant `ES_PE = 'es-PE'`

File: app/scripts/services/Resources.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+      return this.authentication;
+    },
+
+    onWorkerMessage: function onWorkerMessage(event) {
+      if (event.data.status === this.STATUS.AUTHENTICATED) {
+        this.authenticated();
+      } else {
+        this.notAuthenticated();
+        this.authentication = null;
```
Comment (michaelwehrley) [inline app/scripts/services/Resources.js]: does this always run before `authenticate`?  if not, I would consider setting the properties on an initialize function.

File: app/scripts/app.js (hunk 1 above)
Comment (ericcf) [inline app/scripts/app.js]: good call

**Comment by michaelwehrley** [comment]
That was a HUGE PR with some GREAT code.  There is a lot going on in it, and I give you mad props.  Nice work. :+1: Please merge when desired.

File: app/scripts/services/Resources.js (hunk 2 above)
Comment (ericcf) [inline app/scripts/services/Resources.js]: It runs whenever the web worker sends a message. The communication is asynchronous, so each time a message is received, I have to check whether authentication has completed or not. If not, I reset the `Promise`, `resolve`, and `reject` references and start over.

The return value of `authenticate()` is a promise that determines whether you get to the main page or the config page.

Hope this makes sense.

**Pull request merged**